apis/stash/types.go

```diff
@@ -166,3 +131,30 @@ type RetentionPolicy struct {
 	Prune       bool     `json:"prune,omitempty"`
 	DryRun      bool     `json:"dryRun,omitempty"`
 }
+
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Recovery struct {
+	metav1.TypeMeta   `json:",inline,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              RecoverySpec `json:"spec,omitempty"`
+	Status            string       `json:"status,omitempty"`
```

Comment:
RecoveryStatus needs to be redone. See how it is done for Pods: https://github.com/kubernetes/api/blob/409c3b2393cd3359f275bd8b883b4d9c5aec41f6/core/v1/types.go#L2751

```
type RecoveryPhase string 

const(
  ...
  RecoverySucceeded RecoveryPhase = "Succeeded"
)

type RecoveryStatus struct {
	Phase RecoveryPhase `json:"phase,omitempty"`
	Stats []RestoreStats `json:"stats,omitempty"`
}

type RestoreStats struct {
	Path             string       `json:"path,omitempty"`
	Duration       string       `json:"duration,omitempty"`
}
```